### PR TITLE
Utkarsha/refactor migrate api token component

### PR DESCRIPTION
**Pull request by utkarsha-deriv**
## Changes:

Following PR refactors and migrates the API Token Component to Typescript. It also removes the usage of hooks from api package to avoid side-effects. The test cases for the files were also modified accordingly.

### CODE REVIEW

File: packages/account/package.json [hunk 1]
```diff
@@ -36,7 +36,6 @@
         "@deriv/shared": "^1.0.0",
         "@deriv/stores": "^1.0.0",
         "@deriv/translations": "^1.0.0",
-        "@deriv/api": "^1.0.0",
```
Comment (sanjam-deriv) [inline packages/account/package.json:39]: Is this package not being used anywhere else?

File: packages/account/src/Components/api-token/__tests__/api-token-clipboard.spec.tsx [hunk 2]
```diff
@@ -8,7 +8,7 @@ modal_root_el.setAttribute('id', 'modal_root');
 document.body.appendChild(modal_root_el);
 
 describe('ApiTokenClipboard', () => {
-    const mock_props = {
+    const mock_props: React.ComponentProps<typeof ApiTokenClipboard> = {
```
Comment (sanjam-deriv) [inline packages/account/src/Components/api-token/__tests__/api-token-clipboard.spec.tsx:11]: Can we use a direct export here?

File: packages/account/src/Components/api-token/__tests__/api-token-table-row-token-cell.spec.tsx [hunk 3]
```diff
@@ -4,7 +4,7 @@ import userEvent from '@testing-library/user-event';
 import ApiTokenTableRowTokenCell from '../api-token-table-row-token-cell';
 
 describe('ApiTokenTableRowTokenCell', () => {
-    const mock_props = {
+    const mock_props: React.ComponentProps<typeof ApiTokenTableRowTokenCell> = {
```
Comment (sanjam-deriv) [inline packages/account/src/Components/api-token/__tests__/api-token-table-row-token-cell.spec.tsx:7]: Same

File: packages/account/src/Components/api-token/api-token-delete-button.tsx [hunk 4]
```diff
@@ -30,15 +30,17 @@ const ApiTokenDeleteButton = ({ token, popover_alignment = 'left' }: TApiTokenDe
...
 
-    const handleYes = () => {
+    const onSubmit = () => {
         setIsLoading(true);
         deleteToken(token.token);
-        if (isMounted() && isSuccess) {
-            setIsLoading(false);
-            setIsDeleting(false);
-        }
+        deleteToken(token.token).finally(() => {
```
Comment (likhith-deriv) [inline packages/account/src/Components/api-token/api-token-delete-button.tsx]: Why are there 2 consecutive calls for `deleteToken()`? The function returns promise, I think u can remove **Line:37**. and change `onSubmit()` to async and await? Then u need not just have a `finally()`

File: packages/account/src/Components/api-token/__tests__/api-token-clipboard.spec.tsx (hunk 2 above)
Comment (yauheni-deriv) [reply packages/account/src/Components/api-token/__tests__/api-token-clipboard.spec.tsx:11]: @sanjam-deriv , why?
Isn’t it better way to extract component types rather than export them?
if types are not supposed to be reused and to be extended, we better should use this approach I believe. need other opinions on this :)

File: packages/account/src/Components/api-token/__tests__/api-token-table.spec.tsx [hunk 5]
```diff
@@ -21,9 +17,21 @@ describe('ApiTokenTable', () => {
...
         deleteToken: jest.fn(),
-        footer_ref: document.createElement('div'),
-        overlay_ref: document.createElement('div'),
-        toggleOverlay: jest.fn(),
+    };
+    const store = mockStore({
+        ui: {
+            is_mobile: false,
+        },
+    });
```
Comment (yauheni-deriv) [inline packages/account/src/Components/api-token/__tests__/api-token-table.spec.tsx]:
```suggestion
    const store = mockStore({ });
```
it is already false by default. we can omit it here

File: packages/account/src/Components/api-token/api-token-table.tsx [hunk 6]
```diff
@@ -104,14 +107,15 @@ const ApiTokenTable = () => {
...
     return (
         <table className='da-api-token__table'>
             <thead>
                 <tr className='da-api-token__table-header-row'>
-                    <ApiTokenTableRowHeader text={<Localize i18n_default_text='Name' />} />
-                    <ApiTokenTableRowHeader text={<Localize i18n_default_text='Token' />} />
-                    <ApiTokenTableRowHeader text={<Localize i18n_default_text='Scopes' />} />
-                    <ApiTokenTableRowHeader text={<Localize i18n_default_text='Last used' />} />
+                    {api_token_table_header_title.map(title => (
+                        <ApiTokenTableRowHeader key={title} text={<Localize i18n_default_text={title} />} />
```
Comment (likhith-deriv) [inline packages/account/src/Components/api-token/api-token-table.tsx:117]: Does using `<Localize/> `this way? Please re-verify. `<Localize i18n_default_text={'{{text}}'} values={{title} />`. I think dynamic value need to be passed to values prop of Localize component

File: packages/account/src/Sections/Security/ApiToken/api-token.tsx [hunk 7]
```diff
@@ -1,10 +1,9 @@
 import React from 'react';
 import classNames from 'classnames';
 import { Formik, Form, Field, FormikErrors, FieldProps, FormikHelpers } from 'formik';
-import { useApiToken } from '@deriv/api';
```
Comment (likhith-deriv) [inline packages/account/src/Sections/Security/ApiToken/api-token.tsx:4]: Are we removing `useApiToken` hook completely or will it be part of another PR?

File: packages/account/src/Sections/Security/ApiToken/api-token.tsx [hunk 8]
```diff
@@ -43,31 +45,33 @@ const ApiToken = observer(() => {
...
-         * Fetch all API tokens
-         */
-        getApiToken();
-    }, [getApiToken]);
+        getApiTokens();
+
+        return () => {
+            clearTimeout(timeout_ref.current);
+        };
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (likhith-deriv) [inline packages/account/src/Sections/Security/ApiToken/api-token.tsx]: Need to refactor this code. The dependency list shouldn't be suppressed  😅

File: packages/account/package.json (hunk 1 above)
Comment (utkarsha-deriv) [reply packages/account/package.json:39]: This is a duplicate.

File: packages/account/src/Sections/Security/ApiToken/api-token.tsx (hunk 7 above)
Comment (utkarsha-deriv) [reply packages/account/src/Sections/Security/ApiToken/api-token.tsx:4]: We are removing the api hooks entirely for now.

File: packages/account/src/Components/api-token/api-token-table.tsx (hunk 6 above)
Comment (utkarsha-deriv) [reply packages/account/src/Components/api-token/api-token-table.tsx:117]: It's working without passing it as values also.

File: packages/account/src/Components/api-token/__tests__/api-token-table-row-token-cell.spec.tsx (hunk 3 above)
Comment (utkarsha-deriv) [reply packages/account/src/Components/api-token/__tests__/api-token-table-row-token-cell.spec.tsx:7]: This is the usual convention that we use everywhere for types in test case files

File: packages/account/src/Components/api-token/__tests__/api-token-clipboard.spec.tsx (hunk 2 above)
Comment (utkarsha-deriv) [reply packages/account/src/Components/api-token/__tests__/api-token-clipboard.spec.tsx:11]: ++ @yauheni-deriv

File: packages/account/src/Components/api-token/__tests__/api-token-table.spec.tsx [hunk 9]
```diff
@@ -46,25 +54,23 @@ describe('ApiTokenTable', () => {
...
     });
 
     it('should render in mobile view', () => {
+        const new_store_config: TStores = {
+            ...store,
+            ui: {
+                ...store.ui,
+                is_mobile: true,
+            },
+        };
```
Comment (shahzaib-deriv) [inline packages/account/src/Components/api-token/__tests__/api-token-table.spec.tsx]:
```suggestion
        const mock_store = {
            ui: {
                is_mobile: true,
            },
        };
```

It's cleaner to get a new mocked store rather than spreading the old one

Comment (utkarsha-deriv) [reply packages/account/src/Components/api-token/__tests__/api-token-table.spec.tsx]: You mean this yeah? 
```
const mock_store = mockStore({
            ui: {
                is_mobile: true,
            },
 });
```

File: packages/account/src/Components/api-token/api-token-table.tsx (hunk 6 above)
Comment (shahzaib-deriv) [reply packages/account/src/Components/api-token/api-token-table.tsx:117]: This is not the effective way to use this as on build time, anything passed to `i18n_default_text` is extracted by a script and if any string which is not translated yet will be submitted as is i.e. if `Name` isn't translated yet, translation team will get `title` to translate.

And in our case, the extraction will skip this as it's not starting from single or double quotes

Comment (shahzaib-deriv) [reply packages/account/src/Components/api-token/api-token-table.tsx:117]: So, I suggest you can create a function to return the title as `Localize` component by passing the name

something like this:
```suggestion
const api_token_table_header_title = ['name', 'token', 'scopes', 'last_used'];


const getTitle = (key) => {
  const titles = {
  'name': <Localize i18n_default_text="Name"/ >,
   ...
  }
  return titles[key]
}
...
{api_token_table_header_title.map(title => (
     <ApiTokenTableRowHeader key={title} text={<Localize i18n_default_text={getTitle(title)} />} />
))}

File: packages/account/src/Sections/Security/ApiToken/api-token.tsx [hunk 10]
```diff
@@ -43,31 +45,33 @@ const ApiToken = observer(() => {
...
 
+    const timeout_ref = React.useRef<NodeJS.Timeout | undefined>();
+
     React.useEffect(() => {
-        /**
-         * Fetch all API tokens
-         */
-        getApiToken();
-    }, [getApiToken]);
+        getApiTokens();
```
Comment (shahzaib-deriv) [inline packages/account/src/Sections/Security/ApiToken/api-token.tsx:60]: I suggest never do an API call in useEffect without a cancel token or a check if the response is setting any state within the component as it may lead to memory leak problems 

Something like this:

`Can't perform a React state update on an unmounted component.
This is a no-op, but it indicates a memory leak in your application.
To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount
method.`

File: packages/account/src/Components/api-token/api-token-table.tsx (hunk 6 above)
Comment (utkarsha-deriv) [reply packages/account/src/Components/api-token/api-token-table.tsx:117]: But won't this return <Localize i18n_default_text="Name"/ > inside `text={<Localize i18n_default_text={getTitle(title)} />}.` The array already has strings inside so we basically passing them..and the stings are already translated so it works.

File: packages/account/src/Sections/Security/ApiToken/api-token.tsx (hunk 10 above)
Comment (utkarsha-deriv) [reply packages/account/src/Sections/Security/ApiToken/api-token.tsx:60]: I see that in our code we are making api calls in multiple places maybe we can take up this initiative in future refactoring cards. As of now, `Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in the componentWillUnmount method` is not showing up as warning in console. Thanks

File: packages/account/src/Components/api-token/api-token-table.tsx (hunk 6 above)
Comment (shahzaib-deriv) [reply packages/account/src/Components/api-token/api-token-table.tsx:117]: My bad 
I meant like this actually

```
{api_token_table_header_title.map(title => (
	     <ApiTokenTableRowHeader key={title} text={getTitle(title)} />
))}
